pygfunction/api.py

```diff
@@ -0,0 +1,39 @@
+from pygfunction.gfunction import gFunction
+from pygfunction.boreholes import Borehole
+from typing import Dict, Optional, Any, List
+
+
+class PYG(object):
+
+    def __init__(self, borehole_config_list: List[List[float]],
```

Comment:
I think it would be a bit nicer to pass in a list of structures instead of a list of list of floats.  With the list of floats, our tools aren't able to help inform us of what each float should be.  We could do some simple dataclasses that capture the different float members, and even add some helper functions to generate them for common cases.  I just really want this API to be clean and self-documenting as much as possible.